File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsChat.java
package com.github.badoualy.telegram.tl.api;

import com.github.badoualy.telegram.tl.core.TLObject;

/**
 * Abstraction level for the following constructors:
 * <ul>
 * <li>{@link TLChannel}: channel#a14dca52</li>
 * <li>{@link TLChannelForbidden}: channelForbidden#8537784f</li>
 * <li>{@link TLChat}: chat#d91cdd54</li>
 * <li>{@link TLChatEmpty}: chatEmpty#9ba2d800</li>
 * <li>{@link TLChatForbidden}: chatForbidden#7328bdb</li>
 * </ul>
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
public abstract class TLAbsChat extends TLObject {

    protected int id;

    public TLAbsChat() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }
}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsChat.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_BOOLEAN
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_CONSTRUCTOR_ID
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_DOUBLE
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT32
import com.github.badoualy.telegram.tl.TLObjectUtils.SIZE_INT64
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLBytesSerializedSize
import com.github.badoualy.telegram.tl.TLObjectUtils.computeTLStringSerializedSize
import com.github.badoualy.telegram.tl.core.TLObject
import kotlin.Int

/**
 * Abstraction level for the following constructors:
 * * [channel#c88974ac][TLChannel]
 * * [channelForbidden#289da732][TLChannelForbidden]
 * * [chat#d91cdd54][TLChat]
 * * [chatEmpty#9ba2d800][TLChatEmpty]
 * * [chatForbidden#7328bdb][TLChatForbidden]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsChat : TLObject() {
    abstract var id: Int
}
